### PR TITLE
Fix mimisbrunnr battle action

### PR DESCRIPTION
**Pull request by hskim881028**
Fix the problem of stripping the costume when it first entered the Mimisbrunnr

### CODE REVIEW

**Review by boscohyun — APPROVED** [review]
We need some unit tests for this changes.

File: .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs [hunk 1]
```diff
@@ -524,5 +524,50 @@ public void ExecuteThrowFailedAddWorldExceptionWhenDoesNotMimisbrunnr(int alread
...
+        {
+            var avatarState = _initialState.GetAvatarState(_avatarAddress);
+            avatarState.worldInformation = new WorldInformation(
+                0,
+                _tableSheets.WorldSheet,
+                140);
+
+            var costumeId = _tableSheets
+                .CostumeItemSheet
+                .Values
```
Comment (boscohyun) [inline .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs:539]:
```suggestion
                .OrderedList
```

File: .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs [hunk 2]
```diff
@@ -524,5 +524,50 @@ public void ExecuteThrowFailedAddWorldExceptionWhenDoesNotMimisbrunnr(int alread
...
+                .First(r => r.ItemSubType == ItemSubType.FullCostume)
+                .Id;
+            var costume =
+                ItemFactory.CreateItem(_tableSheets.ItemSheet[costumeId], new TestRandom());
+            avatarState.inventory.AddItem(costume);
+
+            var equipmentRow =
+                _tableSheets.EquipmentItemSheet.Values.First(x => x.ElementalType == ElementalType.Fire);
```
Comment (boscohyun) [inline .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs:547]:
```suggestion
                _tableSheets.EquipmentItemSheet.OrderedList.First(x => x.ElementalType == ElementalType.Fire);
```

File: .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs [hunk 3]
```diff
@@ -524,5 +524,50 @@ public void ExecuteThrowFailedAddWorldExceptionWhenDoesNotMimisbrunnr(int alread
...
+            var equipment = ItemFactory.CreateItemUsable(equipmentRow, default, 0);
+            var nextAvatarState = _initialState.SetState(_avatarAddress, avatarState.Serialize());
```
Comment (boscohyun) [inline .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs:549]:
```suggestion
            var nextState = _initialState.SetState(_avatarAddress, avatarState.Serialize());
```

File: Lib9c/Action/MimisbrunnrBattle.cs [hunk 4]
```diff
@@ -189,8 +189,20 @@ public override IAccountStateDelta Execute(IActionContext context)
...
+            
+            var equippableItemstems = avatarState.inventory.Items.Select(i => i.item)
+                .OfType<IEquippableItem>()
+                .Where(x => x.Equipped).ToList();
+            if (equippableItem.Count != equippableItemstems.Count)
+            {
+                throw new InvalidEquipmentException(
+                    "The equipment taken and worn does not match."
+                );
+            }
```
Comment (boscohyun) [inline Lib9c/Action/MimisbrunnrBattle.cs:205]:
```suggestion
```

```
179: avatarState.ValidateEquipments(equipments, context.BlockIndex);
180: avatarState.ValidateConsumable(foods, context.BlockIndex);
181: avatarState.ValidateCostume(costumes);
```

File: .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs [hunk 5]
```diff
@@ -524,5 +524,50 @@ public void ExecuteThrowFailedAddWorldExceptionWhenDoesNotMimisbrunnr(int alread
...
+            };
+
+            Assert.Throws<InvalidEquipmentException>(() =>
+            {
+                action.Execute(new ActionContext
+                {
+                    PreviousStates = nextAvatarState,
+                    Signer = _agentAddress,
+                });
+            });
```
Comment (boscohyun) [inline .Lib9c.Tests/Action/MimisbrunnrBttleTest.cs:570]:
```suggestion
            nextState = action.Execute(new ActionContext
            {
                PreviousStates = nextAvatarState,
                Signer = _agentAddress,
            });
            
            var spawnPlayer = action.Result.FirstOrDefault(e => e is SpawnPlayer);
            Assert.NotNull(spawnPlayer);
            Assert.True(spawnPlayer.Character is Player player);
            // check `player.Costumes`
            // check `player.Equipments`
```

**Review by boscohyun — CHANGES_REQUESTED** [review]

**Review by boscohyun — APPROVED** [review]

**Review by ipdae — APPROVED** [review]

**Review by Unengine — APPROVED** [review]

**Pull request merged**